library/std/src/sys/fs/unix.rs

```diff
@@ -262,7 +262,154 @@ impl ReadDir {
     }
 }
 
-struct Dir(*mut libc::DIR);
+pub struct Dir(*mut libc::DIR);
+
+// dirfd isn't supported everywhere
+#[cfg(not(any(
+    miri,
+    target_os = "redox",
+    target_os = "nto",
+    target_os = "vita",
+    target_os = "hurd",
+    target_os = "espidf",
+    target_os = "horizon",
+    target_os = "vxworks",
+    target_os = "rtems",
+    target_os = "nuttx",
+)))]
+impl Dir {
+    pub fn open<P: AsRef<Path>>(&self, path: P) -> io::Result<File> {
+        let mut opts = OpenOptions::new();
+        opts.read(true);
+        run_path_with_cstr(path.as_ref(), &|path| self.open_c(path, &opts))
+    }
+
+    pub fn open_with<P: AsRef<Path>>(&self, path: P, opts: &OpenOptions) -> io::Result<File> {
+        run_path_with_cstr(path.as_ref(), &|path| self.open_c(path, opts))
+    }
+
+    pub fn open_c(&self, path: &CStr, opts: &OpenOptions) -> io::Result<File> {
+        let flags = libc::O_CLOEXEC
+            | opts.get_access_mode()?
+            | opts.get_creation_mode()?
+            | (opts.custom_flags as c_int & !libc::O_ACCMODE);
+        let fd = cvt_r(|| unsafe {
+            openat64(libc::dirfd(self.0), path.as_ptr(), flags, opts.mode as c_int)
+        })?;
+        Ok(File(unsafe { FileDesc::from_raw_fd(fd) }))
+    }
+
+    // pub fn create_dir<P: AsRef<Path>>(&self, path: P) -> Result<()>
+    // pub fn rename<P: AsRef<Path>, Q: AsRef<Path>>(&self, from: P, to_dir: &Self, to: Q) -> Result<()>
+    // pub fn remove_file<P: AsRef<Path>>(&self, path: P) -> Result<()>
+    // pub fn remove_dir<P: AsRef<Path>>(&self, path: P) -> Result<()>
+    // pub fn symlink<P: AsRef<Path>, Q: AsRef<Path>>(&self, original: P, link: Q)
```

Comment:
It's probably better to delete this, instead of possibly forgetting to remove them later.